include/openenclave/edger8r/common.h

```diff
@@ -90,16 +103,18 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
  * Compute and set the pointer value for the given parameter within the input
  * buffer. Make sure that the buffer has enough space.
  */
-#define OE_SET_IN_POINTER(argname, argsize, argtype)                       \
-    if (pargs_in->argname)                                                 \
-    {                                                                      \
-        pargs_in->argname = (argtype)(input_buffer + input_buffer_offset); \
-        OE_ADD_SIZE(input_buffer_offset, (size_t)(argsize));               \
-        if (input_buffer_offset > input_buffer_size)                       \
-        {                                                                  \
-            _result = OE_BUFFER_TOO_SMALL;                                 \
-            goto done;                                                     \
-        }                                                                  \
+#define OE_SET_IN_POINTER(argname, argcount, argsize, argtype)               \
+    if (_pargs_in->argname)                                                  \
+    {                                                                        \
+        size_t _size;                                                        \
+        OE_COMPUTE_SIZE(argcount, argsize, _size);                           \
```

Comment:
See factoring comment for `OE_COMPUT_SIZE` and `OE_ADD_SIZE` above.

Reply:
Fixed